pgmpy/models/FactorGraph.py

```diff
@@ -133,16 +133,35 @@ def remove_factors(self, *factors):
         >>> from pgmpy.factors import Factor
         >>> G = FactorGraph()
         >>> G.add_nodes_from(['a', 'b', 'c'])
-        >>> G.add_nodes_from(['phi1', 'phi2'])
-        >>> G.add_edges_from([('a', 'phi1'), ('b', 'phi1'),
-        ...                   ('b', 'phi2'), ('c', 'phi2')])
         >>> phi1 = Factor(['a', 'b'], [2, 2], np.random.rand(4))
-        >>> G.add_factors(phi1)
+         >>> G.add_factors(phi1)
         >>> G.remove_factors(phi1)
         """
         for factor in factors:
             self.factors.remove(factor)
 
+    def get_cardinality(self):
+        """
+        >>> from pgmpy.models import FactorGraph
+        >>> from pgmpy.factors import Factor
+        >>> G = FactorGraph()
+        >>> G.add_nodes_from(['a', 'b', 'c'])
+        >>> phi1 = Factor(['a', 'b'], [2, 2], np.random.rand(4))
+        >>> phi2 = Factor(['b', 'c'], [2, 2], np.random.rand(4))
+        >>> G.add_nodes_from([phi1, phi2])
+        >>> G.add_edges_from([('a', phi1), ('b', phi1),
+        ...                   ('b', phi2), ('c', phi2)])
+        >>> G.add_factors(phi1, phi2)
+        >>> G.get_cardinality()
+            defaultdict(<class 'int'>, {'c': 2, 'b': 2, 'a': 2})
+        
+        """
+        cardinalities = defaultdict(int)
```

Comment:
@ankurankan if we don't want to add factors attribute in the <code>UndirectedGraph</code> we can add another derived class from it that can act as a parent class for MarkovModel, FactorGraph and ClusterGraph. I think it can save a lot of redundancy.